map_model/src/pathfind/v2.rs

```diff
@@ -0,0 +1,146 @@
+//! Structures related to the new road-based pathfinding
+//! (https://github.com/a-b-street/abstreet/issues/555) live here. When the transition is done,
+//! things here will probably move into pathfind/mod.rs.
+
+use anyhow::Result;
+
+use crate::{DirectedRoadID, Map, Path, PathRequest, PathStep, TurnID};
+
+/// Transform a sequence of roads representing a path into the current lane-based path, by picking
+/// particular lanes and turns to use.
+pub fn path_v2_to_v1(req: PathRequest, road_steps: Vec<DirectedRoadID>, map: &Map) -> Result<Path> {
+    // This is a somewhat brute-force method: run Dijkstra's algorithm on a graph of lanes and
```

Comment:
I'm not too happy with this approach because of performance, but it works. I'll work next on the 3-tuple approach mentioned below.